unicorn/include/cocaine/unicorn/writable.hpp

```diff
@@ -0,0 +1,73 @@
+/*
+* 2015+ Copyright (c) Anton Matveenko <antmat@yandex-team.ru>
+* All rights reserved.
+*
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*/
+
+#ifndef COCAINE_UNICORN_WRITABLE_ADAPTER_HPP
+#define COCAINE_UNICORN_WRITABLE_ADAPTER_HPP
+
+#include <cocaine/rpc/dispatch.hpp>
+
+namespace cocaine { namespace unicorn {
+
+template<class T>
+class writable_adapter_base_t {
```

Comment:
It's interface used by api. For service concrete is adapter  to deferred and for example for discovery purpose there is a standalone class inhereted from this interface.